### DISCUSSION TITLE
Build OpenRadioss and save binary in a different directory

### DISCUSSION THREAD

**Discussion by bruno-rescale** [OpenRadioss/OpenRadioss#512] [Q&A]
Hi,

I was able to successfully run buil_script.sh and build the starter_linux64_gf binary which appears the SMP  executable. Is there a flag/argument to choose a different location where the binary will be saved? I will need to do the same for building the MPI executable but it seems it uses the same shell script to build it.

I need to save these executables outside of the default OpenRadioss/exec directory. Can the binaries exist outside the source code deck?

**Comment by laurent-altr** [comment]
Hi, 
You can move the starter and the engine binaries. You may also need to move dependencies:

- the `hm_cfg_files` directory and update the `RAD_CFG_PATH` environment variable. 
- the `extlib` directory and update the `LD_LIBRARY_PATH` environment variable

**Comment by roguic** [comment]
Hi,

OpenRadioss uses 2 kinds of executable : the starter and the engine. The starter is a pure smp executable. It reads the input file and creates the restart files for the engine. The engine reads the restart files to compute the simulation. This exec can be a pure smp executable or an hybrid executable. 
The pure smp engine exec is called : engine_linux64_gf
The hybrid engine exec is called : engine_linux64_gf_ompi

The script can generate the 2 execs. The *.o are saved in 2 different folder : cbuild_linux64_gf_ompi for the hybrid engine, cbuild_linux64_gf for the smp exec.

Best,
Romain

**Comment by bruno-rescale** [comment]
Thank you so much for your kind responses.

I do have a follow up question. I'm creating cloud instances which are not persistent and the environments gets setup every time we launch a new cluster. The OS has a native old version of GCC (4.8) and I am automating the preload of GCC (8.4). It's an easy "export PATH=/path/to/gcc/bin" and "export LD_LIBRARY_PATH=/path/to/gcc/lib64" at the top of build_script.sh but i was wondering if there is a way to add such gcc environment paths from the command line as an argument from "build_script.sh".

I tried using the "-addflag" argument but it didn't work:

```
[root@ip-10-23-74-118 /home/centos/OpenRadioss-latest-20220920/starter]$ ./build_script.sh -arch=linux64_gf -addflag="export PATH=/program/openradioss-20220920/gcc/8.4.0/bin:$PATH export LD_LIBRARY_PATH=/program/openradioss-20220920/gcc/8.4.0/lib64:$LD_LIBRARY_PATH"

Build OpenRadioss Starter
Build Arguments :
arch = : linux64_gf
precision = : dp
debug = : 0
static_link = : 0
Addflag : "export PATH=/program/openradioss-20220920/gcc/8.4.0/bin:/program/openradioss-20220920/gcc/8.4.0/bin:/opt/amazon/efa/bin/:/bin:/sbin:/usr/bin:/usr/sbin export LD_LIBRARY_PATH=/program/openradioss-20220920/gcc/8.4.0/lib64:/program/openradioss-20220920/gcc/8.4.0/lib64:/lib:/lib64:/usr/lib:/usr/lib64 "

#threads for Makefile : 1

-- Loading linux64_gf settings
-- modules: /home/centos/OpenRadioss-latest-20220920/starter/cbuild_linux64_gf/CMakeFiles/modules

-- Fortran Compiler: gfortran
-- C Compiler: gcc
-- CPP Compiler: g++
-- Exec name: starter_linux64_gf
-- Flags release:
-- Precision: dp
-- Runtime Static link : 0
-- debug: 0
-- Addflag : export PATH=/program/openradioss-20220920/gcc/8.4.0/bin:/program/openradioss-20220920/gcc/8.4.0/bin:/opt/amazon/efa/bin/:/bin:/sbin:/usr/bin:/usr/sbin export LD_LIBRARY_PATH=/program/openradioss-20220920/gcc/8.4.0/lib64:/program/openradioss-20220920/gcc/8.4.0/lib64:/lib:/lib64:/usr/lib:/usr/lib64
-- Build_type : o

dir : /home/centos/OpenRadioss-latest-20220920/starter/cbuild_linux64_gf/CMakeFiles/includes_starter_linux64_gf added
-- Configuring done
-- Generating done
-- Build files have been written to: /home/centos/OpenRadioss-latest-20220920/starter/cbuild_linux64_gf
Scanning dependencies of target alway_execute
[ 0%] Generating CMakeFiles/includes_starter_linux64_gf/__foo.inc, CMakeFiles/includes_starter_linux64_gf/build_info.inc
build_info.inc
[ 0%] Built target alway_execute
[ 0%] Generating CMakeFiles/includes_starter_linux64_gf/__foo.inc, CMakeFiles/includes_starter_linux64_gf/build_info.inc
build_info.inc
[ 0%] Generating CMakeFiles/includes_starter_linux64_gf/starter_message_description.inc
Scanning dependencies of target starter_linux64_gf
[ 0%] Building Fortran object CMakeFiles/starter_linux64_gf.dir/share/modules1/fvbag_mod.F.o
gfortran: error: export: No such file or directory
gfortran: error: PATH=/program/openradioss-20220920/gcc/8.4.0/bin:/program/openradioss-20220920/gcc/8.4.0/bin:/opt/amazon/efa/bin/:/bin:/sbin:/usr/bin:/usr/sbin: No such file or directory
gfortran: error: export: No such file or directory
gfortran: error: LD_LIBRARY_PATH=/program/openradioss-20220920/gcc/8.4.0/lib64:/program/openradioss-20220920/gcc/8.4.0/lib64:/lib:/lib64:/usr/lib:/usr/lib64: No such file or directory
make[3]: *** [CMakeFiles/starter_linux64_gf.dir/share/modules1/fvbag_mod.F.o] Error 1
make[2]: *** [CMakeFiles/starter_linux64_gf.dir/share/modules1/fvbag_mod.F.o.provides] Error 2
make[1]: *** [CMakeFiles/starter_linux64_gf.dir/all] Error 2
make: *** [all] Error 2```

**Reply by roguic** [reply]
Hi @bsotelo ,

You cannot use -addflag : with -addflag, you add some compiler options to the default compiler flags (for example -addflag="-g -fbacktrace" adds -g option and -fbacktrace option to the ADF variable in the build_script.sh, then this ADF variable is used in the cmake_linux64_gf.txt to define the compiler options).

You can find here (https://stackoverflow.com/questions/17275348/how-to-specify-new-gcc-path-for-cmake) different solutions to your problem.

Best,
Romain

**Reply by bruno-rescale** [reply]
Thanks very much, I am able to solve my issue by correctly loading the environment with "export PATH=/path/to/gcc/bin" and "export LD_LIBRARY_PATH=/path/to/gcc/lib64" before running the compilation script.
